### DISCUSSION TITLE
libUV threads in evo runtime

### DISCUSSION THREAD

**Discussion by Jabadahut50** [evo-lua/evo-runtime#572] [Q&A] [answered]
Evo's documentation notes that it is strictly single threaded but it provides access to the LibUV library which includes multi threading? Am I misunderstanding how LibUV Threads work? Or am I to understand that Evo's runtime itself is inherently single threaded but whatever you write that uses it might not be if they use LibUV threads?

**Comment by rdw-software** [comment, marked as answer]
You're correct. The runtime is inherently single-threaded, meaning your Lua code runs on one thread. Despite that fact, libuv can absolutely make use of multiple threads, so scripts can offload certain tasks like I/O, timers, or other work when needed. Note that newly-spawned threads don't share the same Lua state and therefore don't have access to the main thread's global environment.

**Reply by Jabadahut50** [reply]
Ah excellent... in that case this project is everything I need to base my own project on. You sir have found yourself another user!

**Reply by rdw-software** [reply]
Knock yourself out. But keep in mind that the project is very much experimental, rather immature, and generally unfinished.
